tfhe/src/shortint/ciphertext/compact_list.rs

```diff
@@ -81,29 +91,56 @@ impl CompactCiphertextList {
             par_expand_lwe_compact_ciphertext_list(&mut output_lwe_ciphertext_list, &self.ct_list);
         }
 
-        let pbs_order = match self.expansion_kind {
-            CompactCiphertextListExpansionKind::RequiresCasting => unreachable!(),
-            CompactCiphertextListExpansionKind::NoCasting(pbs_order) => pbs_order,
-        };
-
-        output_lwe_ciphertext_list
-            .as_ref()
-            .chunks_exact(self.ct_list.lwe_size().0)
-            .map(|lwe_data| {
-                let ct = LweCiphertext::from_container(
-                    lwe_data.to_vec(),
-                    self.ct_list.ciphertext_modulus(),
-                );
-                Ciphertext {
-                    ct,
-                    degree: self.degree,
-                    message_modulus: self.message_modulus,
-                    carry_modulus: self.carry_modulus,
-                    pbs_order,
-                    noise_level: self.noise_level,
+        match self.expansion_kind {
```

Comment:
Minor suggestiong to avoid nested matches

```rust
match (self.expansion_kind, casting_mode) {
	(
		CompactCiphertextListExpansionKind::NoCasting(pbs_order),
		_
	) => {
		// without casting
	}
	(
		CompactCiphertextListExpansionKind::RequiresCasting,
		ShortintCompactCiphertextListCastingMode::CastIfNecessary(casting_key)
	) => {
		// no casting
	}
	(
		CompactCiphertextListExpansionKind::RequiresCasting,
		ShortintCompactCiphertextListCastingMode::NoCasting
	) => {
		// error
	}
}
```

Reply:
I feared the double match would complicate the number of cases to keep track, so I manage the error case, do you think the nested match hurt readability ?

Reply:
For me yes, but its a tiny bit